modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentParser.java

```diff
@@ -26,11 +26,26 @@
 final class UserAgentParser {
 
     private final UserAgentCache cache;
+
+    private final DeviceTypeParser deviceTypeParser = new DeviceTypeParser();
+
     private final List<UserAgentSubpattern> uaPatterns = new ArrayList<>();
     private final List<UserAgentSubpattern> osPatterns = new ArrayList<>();
     private final List<UserAgentSubpattern> devicePatterns = new ArrayList<>();
     private final String name;
 
+    UserAgentParser(String name, InputStream regexStream, InputStream deviceTypeRegexStream, UserAgentCache cache) {
+        this.name = name;
+        this.cache = cache;
+
+        try {
+            init(regexStream);
+            this.deviceTypeParser.init(deviceTypeRegexStream);
+        } catch (IOException e) {
+            throw new ElasticsearchParseException("error parsing regular expression file", e);
+        }
+    }
+
     UserAgentParser(String name, InputStream regexStream, UserAgentCache cache) {
```

Comment:
If it is necessary to retain both constructors, delegate to the previous one:
```
UserAgentParser(String name, InputStream regexStream, UserAgentCache cache) {
    this(name, regexStream, null, cache);
}
```
and add an `if` around the initialization of the device type parser:
```
try {
    init(regexStream);
    if (deviceTypeRegexStream != null) {
        deviceTypeParser.init(deviceTypeRegexStream);
    }
} ...
```